### PR TITLE
AP-490: add quoting-platform account-projector

### PR DESCRIPTION
**Pull request by james-milligan**

### CODE REVIEW

File: dev-aws/kafka-shared-msk/account-identity/account.tf [hunk 1]
```diff
@@ -371,3 +371,11 @@ module "cbc_account_events_relay" {
   consume_groups   = ["account-identity.cbc-account-events-relay-v2"]
   cert_common_name = "cbc/cbc-account-events-relay-v2"
 }
+
+# Consume from account-identity.account.unified.events
+module "quoting_platform_account_projector" {
+  source           = "../../../modules/tls-app"
+  cert_common_name = "quoting-platform/account-projector"
+  consume_topics   = [kafka_topic.account_identity_account_unified_events.name]
+  consume_groups   = ["quoting-platform.account-projector"]
```
Comment (mwindlinger) [inline dev-aws/kafka-shared-msk/account-identity/account.tf:380]:
```suggestion
  consume_groups   = ["quoting-platform/account-projector"]
```

File: prod-aws/kafka-shared-msk/account-identity/account.tf [hunk 2]
```diff
@@ -323,3 +323,11 @@ module "cbc_fraud_detection_consumer" {
   consume_groups   = ["account-identity.cbc-fraud-detection-consumer-v1"]
   cert_common_name = "cbc/cbc-fraud-detection-consumer"
 }
+
+# Consume from account-identity.account.unified.events
+module "quoting_platform_account_projector" {
+  source           = "../../../modules/tls-app"
+  cert_common_name = "quoting-platform/account-projector"
+  consume_topics   = [kafka_topic.account_identity_account_unified_events.name]
+  consume_groups   = ["quoting-platform.account-projector"]
```
Comment (mwindlinger) [inline prod-aws/kafka-shared-msk/account-identity/account.tf:332]:
```suggestion
  consume_groups   = ["quoting-platform/account-projector"]
```

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-account-identity
Path: dev-aws/kafka-shared-msk/account-identity
Commit ID: 43577d7e86b01d2f22dedf79ac80c1755d3c482a
Requested At: 2024-10-29T14:14:01Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/account-identity` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: prod-aws
Module: pubsub/kafka-shared-msk-account-identity
Path: prod-aws/kafka-shared-msk/account-identity
Commit ID: 43577d7e86b01d2f22dedf79ac80c1755d3c482a
Requested At: 2024-10-29T14:14:15Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan prod-aws/kafka-shared-msk/account-identity` as comment.

**Review by sbuliarca — APPROVED** [review]

**Comment by uw-infra** [comment]
Terraform run output for
```
Cluster: prod-aws
Module: pubsub/kafka-shared-msk-account-identity
Path: prod-aws/kafka-shared-msk/account-identity
Commit ID: 1950df41539d35d8430c4505fd57c1f90fa77048
```
<details><summary><b>✅ Run Status: Ok, Run Summary: Apply complete! Resources: 2 added, 0 changed, 0 destroyed</b></summary>

```terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.quoting_platform_account_projector.kafka_acl.group_acl["quoting-platform.account-projector"] will be created
  + resource "kafka_acl" "group_acl" {
      + acl_host                     = "*"
      + acl_operation                = "Read"
      + acl_permission_type          = "Allow"
      + acl_principal                = "User:CN=quoting-platform/account-projector"
      + id                           = (known after apply)
      + resource_name                = "quoting-platform.account-projector"
      + resource_pattern_type_filter = "Literal"
      + resource_type                = "Group"
    }

  # module.quoting_platform_account_projector.kafka_acl.topic_acl["account-identity.account.unified.events"] will be created
  + resource "kafka_acl" "topic_acl" {
      + acl_host                     = "*"
      + acl_operation                = "Read"
      + acl_permission_type          = "Allow"
      + acl_principal                = "User:CN=quoting-platform/account-projector"
      + id                           = (known after apply)
      + resource_name                = "account-identity.account.unified.events"
      + resource_pattern_type_filter = "Literal"
      + resource_type                = "Topic"
    }

Plan: 2 to add, 0 to change, 0 to destroy.
module.quoting_platform_account_projector.kafka_acl.group_acl["quoting-platform.account-projector"]: Creating...
module.quoting_platform_account_projector.kafka_acl.topic_acl["account-identity.account.unified.events"]: Creating...
module.quoting_platform_account_projector.kafka_acl.group_acl["quoting-platform.account-projector"]: Creation complete after 0s [id=User:CN=quoting-platform/account-projector|*|Read|Allow|Group|quoting-platform.account-projector|Literal]
module.quoting_platform_account_projector.kafka_acl.topic_acl["account-identity.account.unified.events"]: Creation complete after 0s [id=User:CN=quoting-platform/account-projector|*|Read|Allow|Topic|account-identity.account.unified.events|Literal]

Warning: Argument is deprecated

  with provider["registry.terraform.io/mongey/kafka"],
  on __env.tf line 12, in provider "kafka":
  12: provider "kafka" {

This parameter is now deprecated and will be removed in a later release,
please use `client_cert` instead.

(and one more similar warning elsewhere)

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

```
</details>

> To manually trigger plan again please post `@terraform-applier plan prod-aws/kafka-shared-msk/account-identity` as comment.

**Comment by uw-infra** [comment]
Terraform run output for
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-account-identity
Path: dev-aws/kafka-shared-msk/account-identity
Commit ID: 1950df41539d35d8430c4505fd57c1f90fa77048
```
<details><summary><b>✅ Run Status: Ok, Run Summary: Apply complete! Resources: 2 added, 0 changed, 0 destroyed</b></summary>

```terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.quoting_platform_account_projector.kafka_acl.group_acl["quoting-platform.account-projector"] will be created
  + resource "kafka_acl" "group_acl" {
      + acl_host                     = "*"
      + acl_operation                = "Read"
      + acl_permission_type          = "Allow"
      + acl_principal                = "User:CN=quoting-platform/account-projector"
      + id                           = (known after apply)
      + resource_name                = "quoting-platform.account-projector"
      + resource_pattern_type_filter = "Literal"
      + resource_type                = "Group"
    }

  # module.quoting_platform_account_projector.kafka_acl.topic_acl["account-identity.account.unified.events"] will be created
  + resource "kafka_acl" "topic_acl" {
      + acl_host                     = "*"
      + acl_operation                = "Read"
      + acl_permission_type          = "Allow"
      + acl_principal                = "User:CN=quoting-platform/account-projector"
      + id                           = (known after apply)
      + resource_name                = "account-identity.account.unified.events"
      + resource_pattern_type_filter = "Literal"
      + resource_type                = "Topic"
    }

Plan: 2 to add, 0 to change, 0 to destroy.
module.quoting_platform_account_projector.kafka_acl.topic_acl["account-identity.account.unified.events"]: Creating...
module.quoting_platform_account_projector.kafka_acl.group_acl["quoting-platform.account-projector"]: Creating...
module.quoting_platform_account_projector.kafka_acl.group_acl["quoting-platform.account-projector"]: Creation complete after 0s [id=User:CN=quoting-platform/account-projector|*|Read|Allow|Group|quoting-platform.account-projector|Literal]
module.quoting_platform_account_projector.kafka_acl.topic_acl["account-identity.account.unified.events"]: Creation complete after 0s [id=User:CN=quoting-platform/account-projector|*|Read|Allow|Topic|account-identity.account.unified.events|Literal]

Warning: Argument is deprecated

  with provider["registry.terraform.io/mongey/kafka"],
  on __env.tf line 12, in provider "kafka":
  12: provider "kafka" {

This parameter is now deprecated and will be removed in a later release,
please use `client_cert` instead.

(and one more similar warning elsewhere)

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

```
</details>

> To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/account-identity` as comment.

**Pull request merged**